File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
       HashMap<Integer, Integer> map = new HashMap<>();
       int[] resultingArray = new int[2];

       for(int i = 0; i < numbers.length; i ++) {
           int currentElement = numbers[i];
           int difference = target - currentElement;

           if(!map.containsKey(difference)) {
               map.put(currentElement, i);
           } else {
               return new int[] {map.get(difference) + 1, i + 1};
           }
       }
       
       return resultingArray;
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        hashMap = {}

        for i, num in enumerate(numbers):
            diff = target - num

            if diff in hashMap:
                return [hashMap[diff] + 1, i + 1]
            
            hashMap[num] = i
        